scylla/src/transport/connection.rs

```diff
@@ -820,6 +822,59 @@ impl Connection {
         }
     }
 
+    async fn prepare_batch<'b>(
+        &self,
+        init_batch: &'b Batch,
+        values: impl BatchValues,
+    ) -> Result<Cow<'b, Batch>, QueryError> {
+        let mut to_prepare = HashSet::<String>::new();
```

Comment:
Nit: you can probably use `&str` as the key type. The strings will be borrowed from the `init_batch` object which is alive for the whole duration of this function and we don't modify it, so lifetimes should check out, This will save you an allocation when inserting to the hashset.